### ISSUE TITLE
<Router basename="/main"> is not able to match the URL "/" because it does not start with the basename, so the <Router> won't render anything. warning 

### ISSUE THREAD

**Issue by piszczu4** [daltonmenezes/electron-router-dom#29]
I've got error when trying to run it like below:

```
import { createElectronRouter } from 'electron-router-dom'

export const { Router, registerRoute } = createElectronRouter({
  port: 5123,
  types: {
    /**
     * The ids of the windows of your application, think of these ids as the basenames of the routes
     * this new way will allow your editor's intelisense to help you know which ids are available to use
     * both in the main and renderer process
     */
    ids: ['main'],
  },
})
```

```
main.ts

app.on("ready", () => {
	const mainWindow = new BrowserWindow({
		webPreferences: {
			preload: getPreloadPath(),
		},
		frame: true,
	});

	registerRoute({
		id: 'main',
		browserWindow: mainWindow,
		htmlFile: path.join(__dirname, '../../index.html'),
	  })

	  
	});

```

```
main.tsx

ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <React.StrictMode>
    <AppRoutes />
  </React.StrictMode>
);

```

```
routes.tsx

import { Router } from "../lib/electron-router-dom";
import { Route } from "react-router-dom";
import App from "./App";
import { Games } from "./pages/games";
import { LoginPage } from "./pages/login";

export function AppRoutes() {
  return (
    <Router
      main={
        <>
          <Route path="/" element={<App />} />
          <Route path="/games" element={<Games />} />
          <Route path="/login" element={<LoginPage />} />
        </>
      }
    />
  );
}




```

What am I doing wrong?

**Comment by negreanucalinadrian** [comment]
I am also facing this issue

**Comment by oksr** [comment]
someone able to overcome this one?
I'm also facing that issue

**Comment by negreanucalinadrian** [comment]
I used [electron-vite](https://electron-vite.org) with react-router-dom and worked like a charm

**Comment by daltonmenezes** [comment]
@piszczu4 @negreanucalinadrian @oksr Could you please provide a reproducible repository/example of this problem so I can test and check what's wrong?

**Comment by angelopedroso** [comment]
Same thing, I even tried copy the example, but nothing.

**Comment by daltonmenezes** [comment]
@angelopedroso As you can see, the basic example is working as expected. Without you providing a sample repository simulating the issue you are having, I unfortunately won't be able to help you! Only then will I be able to find out what you might be doing incorrectly.

https://github.com/user-attachments/assets/6a59ddcc-e8d1-4bf5-93be-604ccd43fd57

**Comment by angelopedroso** [comment]
> [@angelopedroso](https://github.com/angelopedroso) As you can see, the basic example is working as expected. Without you providing a sample repository simulating the issue you are having, I unfortunately won't be able to help you! Only then will I be able to find out what you might be doing incorrectly.
> 
>  basic-example-running.mp4

https://github.com/angelopedroso/rotion-electron-test here the code

**Comment by angelopedroso** [comment]
Ok, @daltonmenezes, is something related to:

```ts
  if (is.dev && process.env.ELECTRON_RENDERER_URL) {
    mainWindow.loadURL(process.env.ELECTRON_RENDERER_URL)
  } else {
    mainWindow.loadFile(path.join(__dirname, '../renderer/index.html'))
  }
```

Is necessary remove it to use this lib?

**Comment by daltonmenezes** [comment]
> Ok, [@daltonmenezes](https://github.com/daltonmenezes), is something related with:
> 
>   if (is.dev && process.env.ELECTRON_RENDERER_URL) {
>     mainWindow.loadURL(process.env.ELECTRON_RENDERER_URL)
>   } else {
>     mainWindow.loadFile(path.join(__dirname, '../renderer/index.html'))
>   }
> Is necessary remove it to use this lib?

that's right, `Electron Router DOM` already manages the `loadURL` and `loadFile` logic, so you must remove that code.

**Comment by daltonmenezes** [comment]
> Ok, [@daltonmenezes](https://github.com/daltonmenezes), is something related with:
> 
>   if (is.dev && process.env.ELECTRON_RENDERER_URL) {
>     mainWindow.loadURL(process.env.ELECTRON_RENDERER_URL)
>   } else {
>     mainWindow.loadFile(path.join(__dirname, '../renderer/index.html'))
>   }
> Is necessary remove it to use this lib?

@angelopedroso that's right, `Electron Router DOM` already manages the `loadURL` and `loadFile` logic, so you must remove that code.

**Comment by daltonmenezes** [comment]
Hello everyone, I'm just stopping by to say a few things:

1. The info that @angelopedroso brought is probably the same reason and solution for what you are experiencing. I took the opportunity to update the [introduction page in the docs](https://electron-router-dom.daltonmenezes.com/docs#creating-your-first-routing) to highlight this so that more people do not experience this.
 ![Image](https://github.com/user-attachments/assets/af619f20-b0ed-48a6-acd0-bc75b24a51df)

2. I'll be closing this issue for now, in case anyone else faces this problem again, now we have an issue template in the repository to better obtain info and a link to a reproducible issue test repository. This will help us better to resolve any problems that arise, since I was unable to simulate the reported issue in my tests.

**Comment by Victor-Palha** [comment]
I was facing the same issue.  
I think it would be better if the warning about not using `loadURL` were written at the end:  

"Please delete the following from the `createWindow` function:  

```js
if (is.dev && process.env.ELECTRON_RENDERER_URL) {
    mainWindow.loadURL(process.env.ELECTRON_RENDERER_URL);
} else {
    mainWindow.loadFile(path.join(__dirname, '../renderer/index.html'));
}
```
To avoid error"
